tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -638,3 +638,50 @@ void TTreeIndex::SetTree(TTree *T)
    fTree = T;
 }
 
+/// \cond HIDDEN_SYMBOLS
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Create a deep copy of the TTreeIndex
+/// The new index is not attached to any tree. It is the responsibility of the
+/// caller to attach it to a tree if necessary.
+std::unique_ptr<TVirtualIndex> TTreeIndex::MakeIndexCopy() const
+{
+   auto index = std::make_unique<TTreeIndex>();
+
+   if (fN == 0)
+      return index;
+
+   index->fMajorName = fMajorName;
+   index->fMinorName = fMinorName;
+   index->fN = fN;
+
+   if (!fIndexValues) {
+      index->fIndexValues = nullptr;
+   } else {
+      index->fIndexValues = new Long64_t[index->fN];
+      std::copy(fIndexValues, fIndexValues + fN, index->fIndexValues);
+   }
+
+   if (!fIndexValuesMinor) {
+      index->fIndexValuesMinor = nullptr;
+   } else {
+      index->fIndexValuesMinor = new Long64_t[index->fN];
+      std::copy(fIndexValuesMinor, fIndexValuesMinor + fN, index->fIndexValuesMinor);
+   }
+
+   if (!fIndex) {
+      index->fIndex = nullptr;
+   } else {
+      index->fIndex = new Long64_t[index->fN];
+      std::copy(fIndex, fIndex + fN, index->fIndex);
+   }
+
+   index->fMajorFormula = (fMajorFormula ? static_cast<TTreeFormula *>(fMajorFormula->Clone()) : nullptr);
+   index->fMinorFormula = (fMinorFormula ? static_cast<TTreeFormula *>(fMinorFormula->Clone()) : nullptr);
+   index->fMajorFormulaParent =
+      (fMajorFormulaParent ? static_cast<TTreeFormula *>(fMajorFormulaParent->Clone()) : nullptr);
+   index->fMinorFormulaParent =
+      (fMinorFormulaParent ? static_cast<TTreeFormula *>(fMinorFormulaParent->Clone()) : nullptr);
```

Comment:
Since the `TTreeIndex` constructor already sets them to `nullptr`, perhaps it's simpler to just conditionally set them to the clones.